### PR TITLE
Align RISE with AISE

### PR DESCRIPTION
**Pull request by negvet**

### CODE REVIEW

File: openvino_xai/methods/black_box/rise.py [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
-from openvino_xai.methods.black_box.base import BlackBoxXAIMethod
+from openvino_xai.common.utils import IdentityPreprocessFN, is_bhwc_layout, scaling
+from openvino_xai.methods.black_box.base import BlackBoxXAIMethod, Preset
 
 
 class RISE(BlackBoxXAIMethod):
-    """RISE explains classification models in black-box mode using RISE (https://arxiv.org/abs/1806.07421).
+    """RISE explains classification models in black-box mode using
+    RISE: Randomized Input Sampling for Explanation of Black-box Models
+    (https://arxiv.org/abs/1806.07421).
```
Comment (GalyaZalesskaya) [inline openvino_xai/methods/black_box/rise.py]:
```suggestion
    """RISE explains classification models in black-box mode using
    'RISE: Randomized Input Sampling for Explanation of Black-box Models' paper
    (https://arxiv.org/abs/1806.07421).
```

File: openvino_xai/methods/black_box/rise.py [hunk 2]
```diff
@@ -45,20 +47,23 @@ def __init__(
...
         """
         Generates inference result of the RISE algorithm.
 
         :param data: Input image.
         :type data: np.ndarray
-        :param explain_target_indices: List of target indices to explain.
-        :type explain_target_indices: List[int]
+        :param target_indices: List of target indices to explain.
+        :type target_indices: List[int]
+        :param preset: Speed-Quality preset, defines predefined configurations that manage speed-quality tradeoff.
```
Comment (GalyaZalesskaya) [inline openvino_xai/methods/black_box/rise.py:65]:
```suggestion
        :param preset: Speed-Quality preset, defines predefined configurations that manage the speed-quality tradeoff.
```

File: openvino_xai/methods/black_box/rise.py [hunk 3]
```diff
@@ -74,20 +79,44 @@ def generate_saliency_map(
...
+            return num_masks
+
+        if preset == Preset.SPEED:
+            return 2000
+        elif preset == Preset.BALANCE:
+            return 5000
+        elif preset == Preset.QUALITY:
+            return 8000
+        else:
+            raise ValueError(f"Preset {preset} is not supported.")
```
Comment (GalyaZalesskaya) [inline openvino_xai/methods/black_box/rise.py:119]: What do you think, do we need to proportionally update `num_cells` as well? I remember, you did the research that showed better saliency maps if `num_cels` and `num_iterations` were proportional

File: openvino_xai/methods/black_box/aise.py [hunk 4]
```diff
@@ -124,28 +128,29 @@ def generate_saliency_map(  # type: ignore
...
         return saliency_maps
 
+    @staticmethod
     def _preset_parameters(
-        self,
         preset: Preset,
         num_iterations_per_kernel: int | None,
         kernel_widths: List[float] | np.ndarray | None,
-    ) -> None:
+    ) -> Tuple[int, np.ndarray]:
```
Comment (GalyaZalesskaya) [inline openvino_xai/methods/black_box/aise.py:136]: I have a minor concern here that we return a tuple of unnamed parameters, taking `num_iterations_per_kernel` and transforming it to just `iterations`, `kernel_widths` -> `widths`.
To keep the consistency between input and output, we can rewrite the last part as 

```
        if num_iterations_per_kernel is None:
            num_iterations_per_kernel = iterations 
        if kernel_widths is None:
            kernel_widths = widths
        return num_iterations_per_kernel, kernel_widths
```

**Review by GalyaZalesskaya — COMMENTED** [review]
Thank you for the update!

File: openvino_xai/methods/black_box/rise.py (hunk 3 above)
Comment (negvet) [reply openvino_xai/methods/black_box/rise.py:119]: This is a good point, thanks for noticing. This is correct. But I am not ready to set the exact numbers for now. Let's wait for the metric support and then we can do an ablations study

File: openvino_xai/methods/black_box/aise.py (hunk 4 above)
Comment (negvet) [reply openvino_xai/methods/black_box/aise.py:136]: Good point, updated!

File: openvino_xai/methods/black_box/rise.py (hunk 3 above)
Comment (negvet) [reply openvino_xai/methods/black_box/rise.py:119]: frankly speaking, there are many potential updates possible: run at different seed and aggregate, run at different num_cells and aggregate, etc.

Comment (GalyaZalesskaya) [reply openvino_xai/methods/black_box/rise.py:119]: Can you add a todo note for that please?

**Review by GalyaZalesskaya — APPROVED** [review]

**Pull request merged**